Makefile

```diff
@@ -5,6 +5,7 @@ CLI_VERSION=$(shell grep '^__version__' src/globus_cli/version.py | cut -d '"' -
 	virtualenv --python=$(PYTHON_VERSION) .venv
 	.venv/bin/pip install -U pip setuptools
 	.venv/bin/pip install -e '.[development]'
+	.venv/bin/pip install -e '.[test]'
```

Comment:
I forgot this part of the Makefile even existed.
I may want to suggest removing it later.

Reply:
What alternative path would you suggest for contributers setting up their dev environment?

Reply:
It's mostly OT from this PR, but it depends on what flavor of "dev environment" you mean. For a runnable local copy of the CLI as you work on it, I'd suggest
```shell
python -m venv .venv
.venv/bin/pip install -e .
. .venv/bin/activate
```
as probably all you need.

For test dependencies, I'd typically tell people to rely on `tox`, but of course that fails to acknowledge workflow tools and IDEs which want a test environment like VSCode and PyCharm.

Maybe this is the right place for this to live? But then it needs to be maintained and we should get rid of `.[development]` which is now wrong (and which I think `pip` is just polite about ignoring).

The fact that nothing in CI is using this means it's very easy for it to drift out of date or get broken. If it's supported, even just as a dev workflow, I think we should think about testing it in CI.

Reply:
Sure, considering it non-blocking.

I'm open to testing it in CI but I'm pretty opposed to the idea of taking away a contributor's one-click "get your environment set up to start contributing."

And yeah, I added `[test]` in this situation specifically because my IDE wants an environment that includes our test dependencies if I'm gonna add unit tests. I'm open to driving that virtualenv creation through tox or whatever tool makes the most sense but I'm not open to removing it.